graf3d/eve/inc/TEveVector.h

```diff
@@ -24,15 +26,23 @@ class TVector3;
 template <typename TT>
 class TEveVectorT
 {
+   /// The std::array guarantees that the data is contiguous in memory and we can implement safely Arr().
+   std::array<TT, 3> fXYZ {{0}};
 public:
-   TT fX, fY, fZ; // Components of the vector.
+   TT &fX = *&fXYZ[0], fY = *&fXYZ[1], fZ = *&fXYZ[3]; // Components of the vector.
```

Comment:
I reviewed the code ... took me quite a while, an honest attempt at removing the Arr(), operator[], and automatic casts to float*. The thing really breaks in arcane usages of this in GL tesselator callbacks used in projected objects.

What I could to instead is change fX,fY,fZ to fV[3] and provide x(),y(),z() accessors. This would still touch a lot of eve files.

Note that these are POD classes, so no virtual thingies that are mentioned in your quote above. Do you really believe any architecture will ever align consecutive 32-bit floats on 64-bit boundaries? And if it does, wouldn't it do the same for arrays :)